boot/zephyr/CMakeLists.txt

```diff
@@ -104,8 +104,20 @@ if(CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256)
   target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/sha256.c")
   target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/utils.c")
 
+  # When building for ECDSA, we use our own copy of mbedTLS, so the Zephyr
+  # one must not be enabled or the MBEDTLS_CONFIG_FILE macros will collide.
+  if(DEFINED CONFIG_MBEDTLS)
```

Comment:
Note that this happens when you build with the default `prj.conf` and then use `menuconfig` to switch from RSA to ECDSA. This is a consequence of Kconfig parsing and cannot be fixed by modifying the Kconfig tree. So instead we give a clear error message to the user.

Reply:
The only real fix here is to use 2 different `.conf` files:
* `prj-rsa.conf`
* `prj-ecdsa.conf`

Reply:
It was two files before, I think it makes more sense.

Reply:
OK, this is still relevant if we have 2 files in case someone goes and sets `CONFIG_MBEDTLS=y` in the `prj-ecdsa.conf`, so I can submit a PR after this one to split into 2